File: C#Integration/PokerProtocol/Commands/ListTableCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using PokerProtocol.Commands.Lobby.Response;
using EricUtility;

namespace PokerProtocol.Commands.Lobby
{
    public class ListTableCommand : AbstractCommand
    {
        protected override string CommandName
        {
            get { return ListTableCommand.COMMAND_NAME; }
        }
        public static string COMMAND_NAME = "lobbyLIST_TABLES";


        public ListTableCommand(StringTokenizer argsToken)
        {
        }

        public ListTableCommand()
        {
        }

        public string EncodeResponse(List<TupleTableInfo> tables)
        {
            return new ListTableResponse(this, tables).Encode();
        }
    }
}

File: trunk/JAVA/PokerProtocol/src/protocol/commands/lobby/ListTableCommand.java
package protocol.commands.lobby;

import java.util.ArrayList;
import java.util.StringTokenizer;

import protocol.TupleTableInfo;
import protocol.commands.Command;
import protocol.commands.ICommand;
import protocol.commands.lobby.response.ListTableResponse;

public class ListTableCommand implements ICommand
{
    public static String COMMAND_NAME = "lobbyLIST_TABLES";
    
    public ListTableCommand(StringTokenizer argsToken)
    {
    }
    
    public ListTableCommand()
    {
    }
    
    @Override
    public String encodeCommand()
    {
        final StringBuilder sb = new StringBuilder();
        sb.append(ListTableCommand.COMMAND_NAME);
        sb.append(Command.L_DELIMITER);
        return sb.toString();
    }
    
    public String encodeResponse(ArrayList<TupleTableInfo> tables)
    {
        return new ListTableResponse(tables).encodeCommand();
    }
}
